src/Illuminate/Http/Concerns/InteractsWithInput.php

```diff
@@ -176,6 +176,20 @@ public function all($keys = null)
     }
 
     /**
+     * Get all of the input and files for the request without null input.
+     *
+     * @param  array|mixed $keys
+     * @return array
+     */
+    public function allWithoutNull($keys = null)
+    {
+        $inputs = $this->all($keys);
+        return Arr::whereRecursive($inputs, function ($value, $key) {
+            return !(null === $value || $value === '' || $value === []);
```

Comment:
Empty strings are already being cast to `null` so I think the `$value === ''` check is redundant.